voicevox_engine/tts_pipeline/njd_feature_processor.py

```diff
@@ -0,0 +1,101 @@
+"""NJD Featureの処理"""
+
+import re
+from dataclasses import asdict, dataclass
+
+import pyopenjtalk
+
+from ..utility.character_width_utility import (
+    convert_zenkaku_alphabet_to_hankaku,
+    is_hankaku_alphabet,
+)
+from .katakana_english import (
+    convert_english_to_katakana,
+    should_convert_english_to_katakana,
+)
+
+
+@dataclass
+class NjdFeature:
+    """NJDのFeature"""
+
+    string: str
+    pos: str
+    pos_group1: str
+    pos_group2: str
+    pos_group3: str
+    ctype: str
+    cform: str
+    orig: str
+    read: str
+    pron: str
+    acc: int
+    mora_size: int
+    chain_rule: str
+    chain_flag: int
+
+    @classmethod
+    def from_english_kana(cls, english: str, kana: str) -> "NjdFeature":
+        """英語のカタカナ読みからNjdFeatureを作成する"""
+        # TODO: user_dict/model.py内の処理と重複しているため、リファクタリングする
+        rule_others = (
+            "[イ][ェ]|[ヴ][ャュョ]|[ウクグトド][ゥ]|[テデ][ィェャュョ]|[クグ][ヮ]"
+        )
+        rule_line_i = "[キシチニヒミリギジヂビピ][ェャュョ]|[キニヒミリギビピ][ィ]"
+        rule_line_u = "[クツフヴグ][ァ]|[ウクスツフヴグズ][ィ]|[ウクツフヴグ][ェォ]"
+        rule_one_mora = "[ァ-ヴー]"
+        mora_size = len(
+            re.findall(
+                f"(?:{rule_others}|{rule_line_i}|{rule_line_u}|{rule_one_mora})", kana
+            )
+        )
+        return cls(
+            string=kana,
+            pos="名詞",
+            pos_group1="固有名詞",
+            pos_group2="一般",
+            pos_group3="*",
+            ctype="*",
+            cform="*",
+            orig=english,
+            read=kana,
+            pron=kana,
+            acc=1,
+            mora_size=mora_size,
+            chain_rule="*",
+            chain_flag=-1,
+        )
+
+
+def is_unknown_reading_word(pos: str, chain_rule: str) -> bool:
```

Comment:
ここでしか使わないからローカル関数っぽくしておいてもいいかも
```suggestion
def _is_unknown_reading_word(pos: str, chain_rule: str) -> bool:
```

Reply:
ローカル関数化しました。